src/k8s/pkg/k8sd/pki/k8sdqlite_test.go

```diff
@@ -0,0 +1,180 @@
+9Au1EX7zRzFdrM52YgfBMDyouRKLUjDWCwpxDQIDAQABAoIBADHJBVGR7Q4EEukI
+87Ibm1tS0UDB5DOcRoW4Gmio3BbLGiJLxU2kpBtRjekjFNM9wUkxNOIBW14ZwS1h
+iSr5/XY5Hir/PkRlt0vUdsjQwV9jNlGgYhV7FiF1AtbKRg6XL5kkSjH1oQM8s4bG
+kK8q8Yy4DBQNhkx5/cNDLaNEblFSYOx7fsdfqVH4WLoXguaLgXehwWyK08NkpkaF
+EC47S/X/nqPn4JOwQpVGVBQArun4ATNKLokKZqAIypjtIQl+7V1YEJAIJqbUtLDs
+IuksAXiVISbXdhZm/gUH6Ok9RCcKpJ9VzqraKNA3h+LNgcbJKw+7MijS7Yv3/7/z
+jO90YeECgYEA4J5ufiMa/tNzc4b3TcrNnxcJ1cdDfkxqtq0zv2ByDu46CEHbNI0H
+TCujL4MEwPG1N96apA9IpZcY1rDirIbcFYAMgsYti7cRAqLv5bj8eDnYn174qHHL
+jij7R1hGTcVWBkwzyCaPgCjVaBcW2KwOuEmFxb/SFpfKqY4CEl05WVkCgYEAxAlI
+ykUKLVMY1aoaK48OxmjtgUU7azpgwOaCnu8SOM6mrk6NzMJq7i6P9MgFjfV2UZZc
+qRZK3xGw+Gh3gkhunllJk/2S3QwkeI4NWZGDQT7QhzDQXT43vVlunKHDD/+rpC8r
+CNC2hrAOgPCulipe2HDR9Y4C3WWdnEOeWS8uqtUCgYEAoEGpB6m4SvNGPbifnQsC
+pWzkgXfHucZ/pJHyh6oh9nEVSmriIJ42BKxlozJRI+/PoWra3g5hgHNLL3HIZ9tY
+DqbrRipquHIGWuExU68lwglTenFh65w05NpsXTyn/Di85YVctIJ+g6uehsNic3he
+kDE0liADnkbyOwKsi7mjfxECgYEAmaK1G2DUQwVW900iyXSKndDqIl/B252a6lM9
+l5XB8Cd01jLWSt0rtJNlWu/P+pufKP3wjMvdzcktquEkmERv/UX4tjUK/pZfluOt
+br7t4Rp7jxgglJMIWCtY1wSnvUggmsIktfnsss4T79Ww3htCzdpNkmbDtAPJbAhK
+d5bUuikCgYEAp+mdk4PZ2P2KW5ygXxyAU6Xmc35gLHWrKZqLY9sAnp0mA/G8c1FL
+TXZaEqvvBYG4LEl1QhXukLHE2tP0Azj6+Tg+dhwiPua4OcsZpcPXIBRscris8OgL
+m5cIDhPBuZSCs7ZnhWCHF0WMztl6fqNVp2GuFGbDM+LjAZT2YOdP0Ts=
+-----END RSA PRIVATE KEY-----
+`
+
+	tests := []struct {
+		name          string
+		pki           *K8sDqlitePKI
+		expectedError bool
+	}{
+		{
+			name:          "CompleteCertificates with missing K8sDqliteCert and K8sDqliteKey self-signing not allowed",
+			pki:           &K8sDqlitePKI{},
+			expectedError: true,
+		},
+		{
+			name: "CompleteCertificates with K8sDqliteCert but missing K8sDqliteKey",
+			pki: &K8sDqlitePKI{
+				K8sDqliteCert: cert,
+			},
+			expectedError: true,
+		},
+		{
+			name: "CompleteCertificates with missing K8sDqliteCert but K8sDqliteKey",
+			pki: &K8sDqlitePKI{
+				K8sDqliteKey: key,
+			},
+			expectedError: true,
+		},
+		{
+			name: "CompleteCertificates with K8sDqliteCert and K8sDqliteKey",
+			pki: &K8sDqlitePKI{
+				K8sDqliteCert: cert,
+				K8sDqliteKey:  key,
+			},
+			expectedError: false,
+		},
+		{
+			name: "CompleteCertificates with self-signed CA and successful certificate generation",
+			pki: &K8sDqlitePKI{
+				allowSelfSignedCA: true,
+				hostname:          "localhost",
+				years:             1,
+				ipSANs:            []net.IP{net.ParseIP("127.0.0.1")},
+				dnsSANs:           []string{"localhost"},
+			},
+			expectedError: false,
+		},
+		{
+			name: "CompleteCertificates with self-signed CA allowed",
+			pki: &K8sDqlitePKI{
+				allowSelfSignedCA: true,
+				hostname:          "localhost",
+				years:             1,
+				ipSANs:            []net.IP{},
+				dnsSANs:           []string{"localhost"},
+			},
+			expectedError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := gomega.NewGomegaWithT(t)
```

Comment:
dot-import gomega, so that things don't stutter much
```suggestion
			g := NewWithT(t)
```